test/e2e/specs/editor/plugins/custom-post-types.spec.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Test Custom Post Types', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activatePlugin( 'gutenberg-test-custom-post-types' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin(
+			'gutenberg-test-custom-post-types'
+		);
+	} );
+
+	test( 'should be able to create an hierarchical post without title support', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		await admin.createNewPost( { postType: 'hierar-no-title' } );
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( 'Parent Post' );
+		await editor.publishPost();
+
+		// Create a post that is a child of the previously created post.
+		await admin.createNewPost( { postType: 'hierar-no-title' } );
+		await editor.openDocumentSettingsSidebar();
+		await page
+			.getByRole( 'region', { name: 'Editor settings' } )
+			.getByRole( 'button', {
+				name: 'Hierarchical No Title',
+			} )
+			.click();
+
+		// Open the Document -> Page Attributes panel.
+		await page.getByRole( 'button', { name: 'Page Attributes' } ).click();
+
+		const parentPageLocator = page.getByRole( 'combobox', {
+			name: 'Parent Page',
+		} );
+
+		await parentPageLocator.click();
+		await page.getByRole( 'listbox' ).getByRole( 'option' ).first().click();
```

Comment:
Nit: Could we add some parent locator to make it a bit easier to read here? There could be multiple listboxs on the same page.

Reply:
I wanted to do that, but there's nothing specific besides the "Editor settings" region, and if there is another listbox, it will be in the same region.

I guess the `Combobox` could use some improvements :)

## Screenshot
![CleanShot 2023-04-25 at 13 11 12](https://user-images.githubusercontent.com/240569/234230775-6d08a696-3906-4b42-b8dc-1092b5e5dd6b.png)